mne/preprocessing/nirs/_scalp_coupling_index.py

```diff
@@ -0,0 +1,62 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#          Eric Larson <larson.eric.d@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@inria.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from ...io import BaseRaw
+from ...utils import _validate_type, verbose
+from ..nirs import _channel_frequencies, _check_channels_ordered
+from ...filter import filter_data
+
+
+@verbose
+def scalp_coupling_index(raw, l_freq=0.7, h_freq=1.5,
+                         l_trans_bandwidth=0.3, h_trans_bandwidth=0.3,
+                         verbose=False):
+    r"""Calculate scalp coupling index.
+
+    This function calculates the scalp coupling index [1]. This is a measure of
+    the quality of the connection between the optode and the scalp.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+    %(l_freq)s
+    %(h_freq)s
+    %(l_trans_bandwidth)s
+    %(h_trans_bandwidth)s
+    %(verbose)s
+
+    Returns
+    -------
+    sci : array of float
+        Array containing scalp coupling index for each channel.
+
+    References
+    ----------
+    .. [1] Pollonini, Luca, et al. "Auditory cortex activation to natural
+       speech and simulated cochlear implant speech measured with functional
+       near-infrared spectroscopy." Hearing research 309 (2014): 84-93.
+    """
+    raw = raw.copy().load_data()
+    _validate_type(raw, BaseRaw, 'fnirs_od')
```

Comment:
```suggestion
    _validate_type(raw, BaseRaw, 'raw')
```

Reply:
So according to the journal article the signals should be processed before conversion by beer lambert law (so optical density). But is my method here incorrect?

Reply:
The third arg is just the name of the var to give the user in the error message:

https://github.com/mne-tools/mne-python/blob/master/mne/utils/check.py#L307

So you want to change it to `raw`, because you should make sure the variable named `'raw'` is `BaseRaw` type and give a nice error message if it's not. But what you probably want to do after that is some
```
n_od = pick_types(...)
if not len(n_od):
    raise RuntimeError('... must process optical density...')
```
and/or
```
n_hbx = pick_types(...)
if len(n_hbx):
    raise RuntimeError('... must be run before conversion to HbO/HbR ...')
```